guide/command-handling/adding-features.md

```diff
@@ -386,6 +386,62 @@ No more manually editing your help command! If you aren't completely satisfied w
 :::
 
+## Reloading commands
+
+When writing your commands, you may find it tedious to restart bot every time you want to test even slightest change in your code. However, if you have a command handler, reloading commands can be done with a single command.
+
+Create new command file and paste in the usual format:
+
+```js
+module.exports = {
+	name: 'reload',
+	description: 'Reloads a command',
+	execute(message, args) {
+		// ...
+	},
+};
+```
+
+In this command, you will be using command's name or alias as an only argument. First off, you need to check if command you want to reload exists. This can be done in similiar fashion as getting a command in your main file.  
+Note that you can skip first line if you use [argument checker](/command-handling/adding-features.html#required-arguments) from above:
+
+```js
+if (!args.length) return message.channel.send(`You didn't pass any command to reload, ${message.author}!`);
+const commandName = args[0].toLowerCase();
+const command = message.client.commands.get(commandName)
+	|| message.client.commands.find(cmd => cmd.aliases && cmd.aliases.includes(commandName));
+
+if (!command) return message.channel.send(`There is no command with name or alias \`${commandName}\`, ${message.author}!`);
+```
+
+`message.client`? Yes! Every library specific object has `client` as a property. That means that you don't have to pass it as a parameter to commands to access for example `client.guilds` or `client.commands` which you will need now.
+
+Now, in theory, all there is to do, is to delete the previous command from `client.commands` and require the file again. In practice though, you cannot do this that easy, as `require()` caches the file. If you would require it again, you would simply load previously cached file without any of your changes. In order to remove the file from the cache, you need to add the following line to your command's code:
+
+```js
+delete require.cache[require.resolve(`./${commandName}.js`)];
+```
+
+After removing file from cache, all you have to do is require the file and add freshly loaded command to `client.commands`:
+
+```js
+try {
+	const newCommand = require(`./${commandName}.js`);
+	message.client.commands.set(newCommand.name, newCommand);
+} catch (error) {
+	console.log(error);
+	message.channel.send(`There was an error while reloading a command \`${commandName}\`:\n\`${error.message}\``);
+}
+```
+
+Snippet above uses `try/catch` to load the command file and add it to `client.commands`, and in case of error - log whole error to console and notify about the fact sending only short `error.message`. Note the fact that we never actually delete the command from commands collection, and instead just overwrite it. This prevents from deleting a command, and ending up with no such command in collection upon failed `require()`, as each use of the reload command checks that collection again.
```

Comment:
```suggestion
The snippet above uses a `try/catch` block to load the command file and add it to `client.commands`. In case of an error it will log the full error to the console and notify the user about it with the errors message component `error.message`. Note that you never actually delete the command from the commands collection and instead just overwrite it. This prevents you from deleting a command and ending up with a failed `require()` call, as each use of the reload command checks that collection again.
```
> as each use of the reload command checks that collection again.

I'm not entirely sure what you want to say here, but to me that sentence doesn't make a lot of sense as is.

Reply:
Command handler in my bot breaks when i for example have a syntax error in the file i reload, so the command is not anymore in that collection, and every other use of command checks the collection, and not if you actually have such file.

Reply:
Thanks for the clarification, yeah, that indeed makes sense!